core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -208,12 +209,31 @@ class WorldScreen(val viewingCiv:CivilizationInfo) : CameraStageBaseScreen() {
                 viewingCiv.tradeRequests.isNotEmpty() -> TradePopup(this)
             }
         }
+
+         if  (canAutoCompleteTurn()) {
+            nextTurn()
+         }
+
         updateNextTurnButton(hasVisibleDialogs()) // This must be before the notifications update, since its position is based on it
         notificationsScroll.update(viewingCiv.notifications)
         notificationsScroll.setPosition(stage.width - notificationsScroll.width - 5f,
                 nextTurnButton.y - notificationsScroll.height - 5f)
     }
 
+    private fun canAutoCompleteTurn(): Boolean {
+        return  !hasVisibleDialogs() &&
+                isPlayersTurn &&
+                !waitingForAutosave &&
+                UncivGame.Current.settings.autoCompleteTurn &&
+                !viewingCiv.policies.canAdoptPolicy() &&
+                viewingCiv.greatPeople.freeGreatPeople == 0 &&
+                viewingCiv.popupAlerts.none() &&
+                viewingCiv.tradeRequests.isEmpty() &&
+                !viewingCiv.shouldOpenTechPicker() &&
+                viewingCiv.getIdleUnits().isEmpty() &&
+                stage.actors.last() !is WorldScreenMenuTable
+    }
```

Comment:
Instead of `!viewingCiv.policies.canAdoptPolicy()` I think you should check for `!viewingCiv.policies.shouldOpenPolicyPicker` (so that the autoturn can go on if the player knows he can adopt a policy and refuses to).

Instead of `viewingCiv.getIdleUnits().isEmpty()` you should call `viewingCiv.shouldGoToDueUnit()` (there is an option to ignore the idle units).